agent/tests/test_monolog.c

```diff
@@ -0,0 +1,333 @@
+        "}";
+
+  TEST_ATTRIBUTES_CREATION(context_data, expected_json);
+
+  /* test global and context_data include/exclude rules */
+  nr_attribute_config_destroy(&(NRPRG(txn)->attribute_config));
+  NRPRG(txn)->attribute_config = nr_attribute_config_copy(orig_config);
+  nr_attribute_config_enable_destinations(txn->attribute_config,
+                                          NR_ATTRIBUTE_DESTINATION_LOG);
+  nr_attribute_config_modify_destinations(NRPRG(txn)->attribute_config,
+                                          "context.d*",
+                                          NR_ATTRIBUTE_DESTINATION_LOG, 0);
+  nr_attribute_config_modify_destinations(NRPRG(txn)->attribute_config,
+                                          "context.i*",
+                                          NR_ATTRIBUTE_DESTINATION_LOG, 0);
+  nr_attribute_config_modify_destinations(NRPRG(txn)->attribute_config,
+                                          "context.true_bool_attr",
+                                          NR_ATTRIBUTE_DESTINATION_LOG, 0);
+  nr_attribute_config_modify_destinations(NRPRG(txn)->attribute_config,
+                                          "context.t*", 0,
+                                          NR_ATTRIBUTE_DESTINATION_ALL);
+  nr_attribute_config_modify_destinations(NRPRG(txn)->attribute_config,
+                                          "context.false_bool_attr", 0,
+                                          NR_ATTRIBUTE_DESTINATION_ALL);
+  expected_json
+      = "{"
+        "\"context.true_bool_attr\":true,"
+        "\"context.int_attr\":1234,"
+        "\"context.double_attr\":3.10000,"
+        "\"context.string_attr\":\"string_value\""
+        "}";
+
+  TEST_ATTRIBUTES_CREATION(context_data, expected_json);
+
+  nr_attribute_config_destroy(&orig_config);
+  nr_php_zval_free(&context_data);
+
+  tlib_php_request_end();
+}
+
+static void test_convert_context_data_to_attributes_bad_params(TSRMLS_D) {
+  tlib_php_request_start();
+
+  /* enable context data destination */
+  nrtxn_t* txn = NRPRG(txn);
+  txn->options.log_forwarding_context_data_enabled = 1;
+  nr_attribute_config_enable_destinations(txn->attribute_config,
+                                          NR_ATTRIBUTE_DESTINATION_LOG);
+
+  nr_attributes_t* attributes
+      = nr_monolog_convert_context_data_to_attributes(NULL);
+
+  tlib_pass_if_null("NULL context yields attributes is NULL", attributes);
+
+  // create an undefined zval - nr_php_zval_alloc() returns undefined
+  zval* z = nr_php_zval_alloc();
+
+  tlib_pass_if_equal("zval is undefined type", IS_UNDEF, Z_TYPE_P(z), int, "%d");
+
+  attributes = nr_monolog_convert_context_data_to_attributes(z);
+
+  tlib_pass_if_null("zval of undefined type yields attributes is NULL",
+                    attributes);
+  nr_php_zval_free(&z);
+
+  tlib_php_request_end();
+}
+
+void test_main(void* p NRUNUSED) {
+#if defined(ZTS) && !defined(PHP7)
+  void*** tsrm_ls = NULL;
+#endif /* ZTS && !PHP7 */
```

Comment:
Should be able to remove this since ZTS & PHP<7 are not supported

Reply:
Fixed in commit [35975ba](https://github.com/newrelic/newrelic-php-agent/commit/35975baa425ae10fd6fec82a429d76c5c8f24685).